serialization.rst

```diff
@@ -86,6 +99,13 @@ Examples
     }
   } => { 'foo' : 42, 'prot' : "wine" }
 
+  ThatArrayClass implements MongoDB\BSON\ArraySerializable {
+    private $elements => [ 1, 2, 3 ],
+    function bsonSerialize() : array {
+        return $this->elements;
+    }
+  } => [ 1, 2, 3 ]
```

Comment:
This section only had one example per class, so I didn't add a second one showing how an associative array would get numerically re-indexed. Let me know if we need that, but I think the paragraph above is clear enough.